components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -1162,6 +1257,40 @@ void triggerPreRevokeListeners(OAuthRevocationRequestDTO
         }
     }
 
+    void triggerPreApplicationTokenRevokeListeners(OAuthAppRevocationRequestDTO revokeRequestDTO,
+                                                   OAuthRevocationResponseDTO revokeRespDTO,
+                                                   List<AccessTokenDO> accessTokenDOs) {
+
+        OAuthEventInterceptor oAuthEventInterceptorProxy = OAuthComponentServiceHolder.getInstance()
+                .getOAuthEventInterceptorProxy();
+        if (oAuthEventInterceptorProxy != null && oAuthEventInterceptorProxy.isEnabled()) {
+            Map<String, Object> paramMap = new HashMap<>();
+            try {
+                oAuthEventInterceptorProxy.onPostTokenRevocationByApplication(revokeRequestDTO, revokeRespDTO,
+                        accessTokenDOs, paramMap);
+            } catch (IdentityOAuth2Exception e) {
+                LOG.error("Error occurred when triggering onPostTokenRevocationByConsumer() " +
+                        "of post revocation listener.", e);
+            }
+        }
+    }
+
+    void triggerPreApplicationTokenRevokeListeners(OAuthAppRevocationRequestDTO revokeRequestDTO) {
+
+        OAuthEventInterceptor oAuthEventInterceptorProxy = OAuthComponentServiceHolder.getInstance()
+                .getOAuthEventInterceptorProxy();
+
+        if (oAuthEventInterceptorProxy != null && oAuthEventInterceptorProxy.isEnabled()) {
+            Map<String, Object> paramMap = new HashMap<>();
+            try {
+                oAuthEventInterceptorProxy.onPreTokenRevocationByApplication(revokeRequestDTO, paramMap);
+            } catch (IdentityOAuth2Exception e) {
+                LOG.error("Error occurred when triggering onPreTokenRevocationByConsumer() " +
+                        "of post revocation listener.", e);
```

Comment:
post or pre?

Reply:
Addressed in [6918b82](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1627/commits/6918b828f239ba766e20f66e6d0af997d891413a)